File: banco_problemas_aplicados/nivel_basico/soluciones/java/Problema03.java
/*
 * Problema 03 - Deteccion de Transitorios Impares
 * int cuentaImpares(int[] arr, int tam)  -> cuenta muestras en INDICES impares (i % 2 != 0)
 * (condicion sobre el indice, no sobre el valor; por eso no se lee el contenido)
 */
import java.util.Scanner;

public class Problema03 {

    static int cuentaImpares(int[] arr, int tam) {
        int contador = 0;
        for (int i = 0; i < tam; i++) {
            if (i % 2 != 0) {
                contador = contador + 1;
            }
        }
        return contador;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese el numero de muestras: ");
        int tam = sc.nextInt();

        int[] arr = new int[tam];
        for (int i = 0; i < tam; i++) {
            System.out.print("Muestra " + i + ": ");
            arr[i] = sc.nextInt();
        }

        System.out.println("Muestras en indices impares: " + cuentaImpares(arr, tam));
        sc.close();
    }
}

File: banco_problemas_aplicados/nivel_basico/soluciones/c/problema03.c
/*
 * Problema 03 - Deteccion de Transitorios Impares
 * int cuentaImpares(int Arr[], int tam)
 * Contexto: analisis de ruido EM. Cuenta las muestras ubicadas en INDICES impares.
 *
 * OJO: aqui la condicion es sobre el INDICE (i % 2 != 0), no sobre el valor.
 * Por eso el contenido del arreglo no se usa: solo importa su posicion.
 * (Ver teoria/02_condicion_sobre_indice_vs_valor.md)
 */
#include <stdio.h>

int cuentaImpares(int arr[], int tam) {
    (void) arr;  /* el arreglo forma parte de la firma pedida, aunque no se lea su contenido */
    int contador = 0;
    for (int i = 0; i < tam; i++) {
        if (i % 2 != 0) {            /* i impar: 1, 3, 5, ... */
            contador = contador + 1;
        }
    }
    return contador;
}

int main(void) {
    int tam;
    printf("Ingrese el numero de muestras: ");
    scanf("%d", &tam);

    int arr[tam];
    for (int i = 0; i < tam; i++) {
        printf("Muestra %d: ", i);
        scanf("%d", &arr[i]);
    }

    printf("Muestras en indices impares: %d\n", cuentaImpares(arr, tam));
    return 0;
}
